compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -4191,6 +4199,66 @@ public BLangNode transform(MatchStatementNode matchStatementNode) {
         matchStatement.pos = getPosition(matchStatementNode);
         return matchStatement;
     }
+    
+    @Override
+    public BLangNode transform(ClientResourceAccessActionNode clientResourceAccessActionNode) {
+        BLangInvocation.BLangResourceAccessInvocation resourceInvocation = TreeBuilder.createResourceAccessInvocation();
+        resourceInvocation.expr = createExpression(clientResourceAccessActionNode.expression());
+
+        SeparatedNodeList<Node> resourceAccessPath = clientResourceAccessActionNode.resourceAccessPath();
+        List<BLangExpression> pathSegments = new ArrayList<>(resourceAccessPath.size());
+        for (Node resourceAccessSegment : resourceAccessPath) {
+            switch (resourceAccessSegment.kind()) {
+                case COMPUTED_RESOURCE_ACCESS_SEGMENT:
+                    pathSegments.add(
+                            createExpression(((ComputedResourceAccessSegmentNode) resourceAccessSegment).expression()));
+                    break;
+                case RESOURCE_ACCESS_REST_SEGMENT:
+                    ResourceAccessRestSegmentNode resourceAccessRestSegment =
+                            (ResourceAccessRestSegmentNode) resourceAccessSegment;
+                    BLangExpression spreadMemberExpr = createSpreadMemberExpr(resourceAccessRestSegment.expression(),
+                            getPosition(resourceAccessRestSegment));
+                    pathSegments.add(spreadMemberExpr);
+                    break;
+                default:
+                    Token resourceSegmentName = (Token) resourceAccessSegment;
+                    pathSegments.add(createStringLiteral(resourceSegmentName.text(), getPosition(resourceSegmentName)));
+            }
+        }
+
+        BLangListConstructorExpr listConstructorExpr = (BLangListConstructorExpr)
+                TreeBuilder.createListConstructorExpressionNode();
+        listConstructorExpr.exprs = pathSegments;
+        if (pathSegments.size() == 0) {
+            listConstructorExpr.pos = getPosition(clientResourceAccessActionNode.slashToken());
+        } else {
+            listConstructorExpr.pos = 
+                    getPosition(resourceAccessPath.get(0), resourceAccessPath.get(pathSegments.size() - 1));
+        }
+        
+        resourceInvocation.resourceAccessPathSegments = listConstructorExpr;
+        
+        if (clientResourceAccessActionNode.methodName().isPresent()) {
+            resourceInvocation.name = createIdentifier(clientResourceAccessActionNode.methodName().get().name());
+        } else {
+            resourceInvocation.name = createIdentifier(null, "get");
```

Comment:
Dont we need to maintain these method names in a constant file?

Reply:
https://github.com/ballerina-platform/ballerina-lang/pull/36600/files#r905797049